window/src/os/wayland/seat.rs

```diff
@@ -65,9 +75,23 @@ impl SeatHandler for WaylandState {
         _conn: &Connection,
         _qh: &QueueHandle<Self>,
         _seat: WlSeat,
-        _capability: smithay_client_toolkit::seat::Capability,
+        capability: Capability,
     ) {
-        todo!()
+        match capability {
+            Capability::Keyboard => {
+                log::trace!("Lost keyboard capability");
+                self.keyboard.take().unwrap().release();
```

Comment:
Please avoid the unwrap for this; I'm assuming that this will work:

```suggestion
                self.keyboard.take().map(|p| p.release());
```